revolv/project/tests/test_integration.py

```diff
@@ -0,0 +1,193 @@
+
+
+class PostProjectUpdatesTest(TestUserMixin, UserTestingMixin, TestCase):
+
+    def assert_user_can_create_project_update(self):
+        """
+        Tests that administrators and ambassadors can create project updates. This assert function logs in, creates a project,
+        posts an update to the project, and then verifies that the posted update exists.
+        """
+        response = self.send_test_user_login_request()
+        self.assertUserAuthed(response)
+        project = Project.factories.base.create()
+        response = self.client.post('/project/%d/update' % project.pk, {'update_text': 'This is an update'})
+        new_updates = ProjectUpdate.objects.filter(update_text = 'This is an update')
+        self.assertEqual(len(new_updates),1)
+        self.assertEqual(new_updates[0].project, project)
+
+    def test_admin_form_submission(self):
+        """
+        Tests that administrators can create project updates.
+        This function makes an administrator and then calls the assert_user_can_create_project_update function.
+        """
+        self.test_profile.make_administrator()
+        self.assert_user_can_create_project_update()
+
+    def test_ambassador_form_submission(self):
+        """
+        Tests that ambassadors can create project updates.
+        This function makes an ambassador and then calls the assert_user_can_create_project_update function.
+        """
+        self.test_profile.make_ambassador()
+        self.assert_user_can_create_project_update()
+
+    def test_donor_form_submission(self):
+        """
+        Tests that donors can't create project updates.
+        This function makes a donor, logs in, creates a project, and then attempts to post an update to the project.
+        """
+        self.test_profile.make_donor()
+        response = self.send_test_user_login_request()
+        self.assertUserAuthed(response)
+        project = Project.factories.base.create()
+        response = self.client.post('/project/%d/update' % project.pk, {'update_text': 'This is an update'})
+        
+        # the deny_access method is called in the view from the userdatamixin, which returns a redirect to the homepage
+        self.assertEqual(response.status_code, 302)
+        
+class EditProjectUpdatesTest(TestUserMixin, UserTestingMixin, TestCase):
+
+    def assert_make_update(self, project, text):
+        """
+        Creates and posts an update with the given text to a given project. 
+        Returns the update that was just created.
+        """
+        response = self.send_test_user_login_request()
+        self.assertUserAuthed(response)
+        #response = self.client.post('/project/%d/update' % project.pk, {'update_text': text})
+        ProjectUpdate(project = project, update_text = text).save()
+        new_updates = ProjectUpdate.objects.filter(update_text = text)
+        return new_updates[0]
+
+    def assert_test_edit_update(self, donor = False):
```

Comment:
This one definitely SHOULD start with `assert_` but having `test` in the name also makes it confusing. How about `assert_user_can_or_cant_make_updates` or `assert_user_update_permissions_work`?